core/src/main/java/de/danoeh/antennapod/core/storage/DBTasks.java

```diff
@@ -388,14 +382,31 @@ public static synchronized Feed updateFeed(Context context, Feed newFeed, boolea
                         savedFeed.getItems().add(idx, item);
                     }
 
-                    // only mark the item new if it was published after or at the same time
-                    // as the most recent item
-                    // (if the most recent date is null then we can assume there are no items
-                    // and this is the first, hence 'new')
-                    // New items that do not have a pubDate set are always marked as new
-                    if (item.getPubDate() == null || priorMostRecentDate == null
+                    // Only send the item to the Inbox if the if the "New Episodes Action"
+                    // preference is set to "Add to Inbox", and one of the following is true:
+                    //
+                    //   1. The item was published after or at the same time as
+                    //      the most recent item
+                    //   2. The most recent date is null (i.e. no previous
+                    //      items and this is the first)
+                    //   3. The item has no pubDate
+                    //
+                    FeedPreferences.NewEpisodesAction action = savedFeed.getPreferences().getNewEpisodesAction();
+                    if (action == FeedPreferences.NewEpisodesAction.GLOBAL) {
+                        // Feed preference is set to global, check if the globlal setting is
+                        // to add to the inbox
+                        String globalNewEpisodesAction = UserPreferences.getNewEpisodesAction();
+                        if (globalNewEpisodesAction.equals(FeedPreferences.NewEpisodesAction.ADD_TO_INBOX_VALUE)) {
```

Comment:
Why is this check needed? Can't you simply set it to the global value in this case?

Reply:
The global preference is a string, and the feed preference is an enum, so I cannot.